### PR TITLE
fix!: hide archived rooms in room.autocomplete endpoint

### PR DESCRIPTION
**Pull request by heitortanoue**
## Proposed changes (including videos or screenshots)

Issue with the create option discussion, the list when you choose the Channel or Group is showing Archived Teams/groups/channels

## Steps to test or reproduce

- Create at least one channel/group/team and then archive them

- Choose to create a discussion and then you will be able to pick this archived channel from the drop down suggestion list for parent channel

## Further comments

SUP-286

### CODE REVIEW

File: apps/meteor/server/models/raw/Rooms.ts [hunk 1]
```diff
@@ -294,7 +294,10 @@ export class RoomsRaw extends BaseRaw<IRoom> implements IRoomsModel {
 				},
 			],
 			prid: { $exists: false },
-			$and: [{ $or: [{ federated: { $exists: false } }, { federated: false }] }],
+			$and: [
+				{ $or: [{ federated: { $exists: false } }, { federated: false }] },
+				{ $or: [{ archived: { $exists: false } }, { archived: false }] },
+			],
```
Comment (matheusbsilva137) [inline apps/meteor/server/models/raw/Rooms.ts:300]:
```suggestion
			$and: [{ federated: { $ne: true } }, { archived: { $ne: true } }],
```
Your suggestion works, but I think it's more concise and readable like this.

**Review by matheusbsilva137 — CHANGES_REQUESTED** [review]

**Review by matheusbsilva137 — COMMENTED** [review]
Also, I think we should consider this a **breaking change** since this changes the output of the `rooms.autocomplete.channelAndPrivate` endpoint. I'll approve it for now since it looks fine to me, but I'll mark it as a major change.
Please tell me if you agree, @MarcosSpessatto

**Review by matheusbsilva137 — DISMISSED** [review]

**Comment by MarcosSpessatto** [comment]
> Also, I think we should consider this a **breaking change** since this changes the output of the `rooms.autocomplete.channelAndPrivate` endpoint. I'll approve it for now since it looks fine to me, but I'll mark it as a major change. Please tell me if you agree, @MarcosSpessatto

Yes, I agree. Let's keep this as a breaking change then. 👍

**Review by MarcosSpessatto — APPROVED** [review]

File: .changeset/three-avocados-search.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@rocket.chat/meteor": major
+---
+
+Fixed archived rooms returned in `rooms.autocomplete.channelAndPrivate` endpoint
```
Comment (ggazzo) [inline .changeset/three-avocados-search.md:5]: the description does not explain to me what changes for this to be a break change

**Review by ggazzo — CHANGES_REQUESTED** [review]

**Review by ggazzo — APPROVED** [review]

**Pull request merged**